src/protagonist/CleanupHandler/AssetUpdatedHandler.cs

```diff
@@ -0,0 +1,388 @@
+        (HashSet<ObjectInBucket> objectsToRemove, HashSet<ObjectInBucket> foldersToRemove) s3Objects)
+    {
+        foreach (var deliveryChannel in modifiedOrAdded)
+        {
+            if (assetBefore.ImageDeliveryChannels.Any(x => x.Channel == deliveryChannel.Channel)) // checks for updated rather than added
+            {
+                await CleanupChangedPolicy(deliveryChannel, assetAfter, s3Objects.objectsToRemove);
+            }
+        }
+    }
+    
+    private async Task CleanupRemoved(ImageDeliveryChannel deliveryChannelRemoved, Asset assetAfter, 
+        (HashSet<ObjectInBucket> objectsToRemove, HashSet<ObjectInBucket> foldersToRemove) s3Objects)
+    {
+        switch (deliveryChannelRemoved.Channel)
+        {
+            case AssetDeliveryChannels.Image:
+                CleanupRemovedImageDeliveryChannel(assetAfter, s3Objects.objectsToRemove);
+                break;
+            case AssetDeliveryChannels.Thumbnails:
+                await CleanupRemovedThumbnailDeliveryChannel(assetAfter, s3Objects);
+                break;
+            case AssetDeliveryChannels.Timebased:
+                await CleanupRemovedTimebasedDeliveryChannel(assetAfter, s3Objects.objectsToRemove);
+                break;
+            case AssetDeliveryChannels.File:
+                CleanupFileDeliveryChannel(assetAfter, s3Objects.objectsToRemove);
+                break;
+            default:
+                logger.LogDebug("policy {PolicyName} does not require any changes for asset {AssetId}",
+                    deliveryChannelRemoved.DeliveryChannelPolicy.Name, assetAfter.Id);
+                break;
+        }
+    }
+
+    private async Task CleanupChangedPolicy(ImageDeliveryChannel deliveryChannelModified, Asset assetAfter, 
+        HashSet<ObjectInBucket> objectsToRemove)
+    {
+        switch (deliveryChannelModified.Channel)
+        {
+            case AssetDeliveryChannels.Image:
+                CleanupChangedImageDeliveryChannel(deliveryChannelModified, assetAfter, objectsToRemove);
+                break;
+            case AssetDeliveryChannels.Thumbnails:
+                await CleanupChangedThumbnailDeliveryChannel(assetAfter, objectsToRemove);
+                break;
+            case AssetDeliveryChannels.Timebased:
+                await CleanupChangedTimebasedDeliveryChannel(deliveryChannelModified, assetAfter, objectsToRemove);
+                break;
+            default:
+                logger.LogDebug("policy {PolicyName} does not require any changes for asset {AssetId}",
+                    deliveryChannelModified.DeliveryChannelPolicy.Name, assetAfter.Id);
+                break;
+        }
+    }
+
+    private async Task CleanupChangedTimebasedDeliveryChannel(ImageDeliveryChannel imageDeliveryChannel, Asset assetAfter, HashSet<ObjectInBucket> objectsToRemove)
+    {
+        var presetList = JsonSerializer.Deserialize<List<string>>(imageDeliveryChannel.DeliveryChannelPolicy.PolicyData);
```

Comment:
is it worth using overload from `DeliveryChannelPolicyX`? Either directly or introducing a new one?

```cs
var presetList = imageDeliveryChannel.DeliveryChannelPolicy.PolicyDataAs<List<string>>();

// or with new method, like .ThumbsDataAsSizeParameters()
var presetList = imageDeliveryChannel.DeliveryChannelPolicy.FooData();
```

benefit of latter, specific method is it incorporates safety check DCP is of correct type and avoids other consumers needing to explicitly know and specify the type each time.